### PR TITLE
RFC 0004-isolate_integrator_modifications_via_porting_interfaces

### PR DESCRIPTION
**Pull request by bsimondi**
As discussed with Michael Habrat and Eric Berry.

### CODE REVIEW

**Review by ericeberry — CHANGES_REQUESTED** [review]
Please change directory to 0004-isolate_integrator_modifications_via_porting_interfaces

**Comment by kihun-amlogic** [comment]
Some code in internal directory has problem with some compiler or platform. In such cases where some modification is necessary, will all of the codes requiring modification be moved to the porting layer?

**Comment by ericeberry** [comment]
Code in the internal directory and the porting directory can be modified. Any code outside these directories should not be modified unless it is absolutely necessary.

From: kihun-amlogic ***@***.***>
Date: Wednesday, April 26, 2023 at 6:15 PM
To: rdkcentral/tasecureapi ***@***.***>
Cc: Berry, Eric ***@***.***>, Comment ***@***.***>
Subject: Re: [rdkcentral/tasecureapi] RFC 0004-isolate_integrator_modifications_via_porting_interfaces (PR #63)

Some code in internal directory has problem with some compiler or platform. In such cases where some modification is necessary, will all of the codes requiring modification be moved to the porting layer?

**Review by kihun-amlogic — DISMISSED** [review]

**Review by LisShen — DISMISSED** [review]

**Review by hchoi-brcm — DISMISSED** [review]

**Review by ericeberry — APPROVED** [review]

File: docs/rfcs/accepted/0004-isolate_integrator_modifications_via_porting_interfaces/README.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+iteration, it also allows to seemlessly update the core functionality with little to no change 
+from partners, increasing the chances of update adoption.
+Finally, having a clean separation of ownership of the inner code, versus what is asked to 
+partners to implement, makes the code review process much easier: no delta should be seen on the 
+code, except on the interfaces that are required, which are the only piece of code that now need 
+to be reviewed.
+
+## Updates/Obsoletes
+
+-
```
Comment (mhabrat) [inline docs/rfcs/accepted/0004-isolate_integrator_modifications_via_porting_interfaces/README.md:26]: As a formality, let's fill in blank sections with "None.".

File: docs/rfcs/accepted/0004-isolate_integrator_modifications_via_porting_interfaces/README.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+## Affected platforms
+
+-
```
Comment (mhabrat) [inline docs/rfcs/accepted/0004-isolate_integrator_modifications_via_porting_interfaces/README.md:30]: Shouldn't this be "All"?

**Review by mhabrat — CHANGES_REQUESTED** [review]

**Review by ericeberry — APPROVED** [review]

File: docs/rfcs/accepted/0004-isolate_integrator_modifications_via_porting_interfaces/README.md (hunk 2 above)
Comment (ericeberry) [reply docs/rfcs/accepted/0004-isolate_integrator_modifications_via_porting_interfaces/README.md]: Done

File: docs/rfcs/accepted/0004-isolate_integrator_modifications_via_porting_interfaces/README.md (hunk 1 above)
Comment (ericeberry) [reply docs/rfcs/accepted/0004-isolate_integrator_modifications_via_porting_interfaces/README.md]: Done

**Review by mhabrat — APPROVED** [review]

**Pull request merged**